src/chat/chatAgentRequestHandler.ts

```diff
@@ -0,0 +1,633 @@
+        const activeEditor = vscode.window.activeTextEditor;
+
+        async function findEditorFromReferences(
+            references: readonly vscode.ChatPromptReference[],
+        ): Promise<vscode.TextEditor | undefined> {
+            const tabGroups = vscode.window.tabGroups.all;
+
+            // Function to check if document is SQL
+            function isSqlDocument(document: vscode.TextDocument): boolean {
+                const sqlLanguageIds = ["sql", "mssql"];
+                return sqlLanguageIds.includes(document.languageId);
+            }
+
+            for (const reference of references) {
+                const value = reference.value;
+                let referenceUri: vscode.Uri | undefined;
+
+                if (value instanceof vscode.Location) {
+                    referenceUri = value.uri;
+                } else if (value instanceof vscode.Uri) {
+                    referenceUri = value;
+                }
+
+                if (referenceUri) {
+                    // Try to find this URI in the visible editors, but only if it's an SQL file
+                    const matchingEditor = vscode.window.visibleTextEditors.find(
+                        (editor) =>
+                            editor.document.uri.toString() === referenceUri?.toString() &&
+                            isSqlDocument(editor.document),
+                    );
+
+                    if (matchingEditor) {
+                        return matchingEditor;
+                    }
+
+                    // Try to find this URI in tab groups
+                    for (const group of tabGroups) {
+                        const activeTab = group.activeTab;
+                        if (activeTab) {
+                            try {
+                                /* eslint-disable @typescript-eslint/no-explicit-any */
+                                const tabUri =
+                                    (activeTab.input as any)?.uri ||
+                                    (activeTab.input as any)?.textEditor?.document?.uri;
+                                /* eslint-enable @typescript-eslint/no-explicit-any */
+
+                                if (tabUri && tabUri.toString() === referenceUri.toString()) {
+                                    const editor = vscode.window.visibleTextEditors.find(
+                                        (ed) => ed.document.uri.toString() === tabUri.toString(),
+                                    );
+                                    // Only return the editor if it's an SQL document
+                                    if (editor && isSqlDocument(editor.document)) {
+                                        return editor;
+                                    }
+                                }
+                            } catch (error) {
+                                console.log("Error accessing tab properties:", error);
+                            }
+                        }
+                    }
```

Comment:
Seems a good opportunity to dedupe against the below search with a helper function that optionally takes the specific target URI

Reply:
I'll consider for follow-up but don't want to refactor now.